File: activity/app/src/main/java/com/rikkei/training/activity/activity/DActivity.java
package com.rikkei.training.activity.activity;

import android.content.Intent;
import android.view.View;
import android.widget.Button;

import com.rikkei.training.activity.R;
import com.rikkei.training.activity.base.BaseActivity;

public class DActivity extends BaseActivity {
    private Button mButtonDE;

    @Override
    protected void initData() {

    }

    @Override
    protected void initViews() {
        mButtonDE = findViewById(R.id.btn_act_de);
        mButtonDE.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(DActivity.this, EActivity.class);
                startActivity(intent);
            }
        });
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_d;
    }
}

File: Activitykotlin/app/src/main/java/com/rikkei/training/activitykotlin/Activity/DActivity.kt
package com.rikkei.training.activitykotlin.Activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.rikkei.training.activitykotlin.R
import kotlinx.android.synthetic.main.activity_d.*

class DActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_d)
        btn_gt_de.setOnClickListener(View.OnClickListener {
            intent = Intent(this, EActivity::class.java)
            startActivity(intent)
        })
    }
}
